### PR TITLE
FOUNDATIONS: UserAccess - Retrieve All Organisations User Has AccesssTo

### PR DESCRIPTION
**Pull request by cjdutoit**

### CODE REVIEW

**Review by JParkesAD — DISMISSED** [review]

File: ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/UserAccesses/UserAccessesTests.Exceptions.RetrieveAllOrganisationsUserHasAccessTo.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        [Fact]
+        public async Task ShouldThrowServiceExceptionOnRetrieveAllOrganisationsUserHasAccessToAsync()
+        {
+            // given
+            Guid randomUserAccessId = Guid.NewGuid();
+            Exception serviceError = new Exception();
+
+            var failedServiceUserAccessException = new FailedServiceUserAccessException(
+                message: "Failed service user access error occurred, contact support.",
+                innerException: serviceError);
```
Comment (MichaelOSullivanAnswer) [inline ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/UserAccesses/UserAccessesTests.Exceptions.RetrieveAllOrganisationsUserHasAccessTo.cs:77]: There's an inconsistency in the indentation on 'innerException' in this test and the one above. I think this way is correct, with 'innerException' in-line with 'message'.

File: ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/UserAccesses/UserAccessesTests.Exceptions.RetrieveAllOrganisationsUserHasAccessTo.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        public async Task ShouldThrowCriticalDependencyExceptionOnRetrieveAllOrganisationsUserHasAccessToAsync()
+        {
+            // given
+            Guid randomUserAccessId = Guid.NewGuid();
+            SqlException sqlException = CreateSqlException();
+
+            var failedUserAccessStorageException =
+                new FailedStorageUserAccessException(
+                    message: "Failed user access storage error occurred, contact support.",
+                        innerException: sqlException);
```
Comment (MichaelOSullivanAnswer) [inline ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/UserAccesses/UserAccessesTests.Exceptions.RetrieveAllOrganisationsUserHasAccessTo.cs:27]: innerException over indented here

File: ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/UserAccesses/UserAccessesTests.Exceptions.RetrieveAllOrganisationsUserHasAccessTo.cs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+            var expectedUserAccessDependencyException =
+                new UserAccessDependencyException(
+                    message: "UserAccess dependency error occurred, contact support.",
+                        innerException: failedUserAccessStorageException);
```
Comment (MichaelOSullivanAnswer) [inline ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/UserAccesses/UserAccessesTests.Exceptions.RetrieveAllOrganisationsUserHasAccessTo.cs:32]: innerException over indented here

File: ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/UserAccesses/UserAccessesTests.Exceptions.RetrieveAllOrganisationsUserHasAccessTo.cs (hunk 1 above)
Comment (cjdutoit) [reply ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/UserAccesses/UserAccessesTests.Exceptions.RetrieveAllOrganisationsUserHasAccessTo.cs:77]: fixed

File: ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/UserAccesses/UserAccessesTests.Exceptions.RetrieveAllOrganisationsUserHasAccessTo.cs (hunk 3 above)
Comment (cjdutoit) [reply ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/UserAccesses/UserAccessesTests.Exceptions.RetrieveAllOrganisationsUserHasAccessTo.cs]: fixed

File: ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/UserAccesses/UserAccessesTests.Exceptions.RetrieveAllOrganisationsUserHasAccessTo.cs (hunk 2 above)
Comment (cjdutoit) [reply ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/UserAccesses/UserAccessesTests.Exceptions.RetrieveAllOrganisationsUserHasAccessTo.cs]: fixed

**Review by MichaelOSullivanAnswer — APPROVED** [review]

**Pull request merged**